eds4jinja2/adapters/namespace_handler.py

```diff
@@ -0,0 +1,106 @@
+    - Shortening of URIs to their QName forms
+"""
+
+import logging
+from pprint import pprint
+from typing import Dict, List
+
+import numpy as np
+import rdflib
+from pandas import DataFrame
+
+from eds4jinja2.adapters import invert_dict
+from eds4jinja2.adapters.prefix_cc_fetcher import prefix_cc_lookup_base_uri
+
+logger = logging.getLogger(__name__)
+
+
+class NamespaceInventory(rdflib.namespace.NamespaceManager):
+
+    def __init__(self, namespace_definition_dict=None):
+        super().__init__(rdflib.Graph())
+
+        if namespace_definition_dict:
+            # reduce the namespace definition dictionary and bind the definitions
+            for prefix, namespace in invert_dict(invert_dict(namespace_definition_dict)).items():
+                self.bind(prefix=prefix, namespace=namespace, replace=True, override=True)
+
+        self._remote_query_cash = []
+
+    def namespaces_as_dict(self):
+        """
+        :return: return the namespace definitions as a dict
+        """
+        return {prefix: ns_uri.toPython() for prefix, ns_uri in self.namespaces()}
+
+    def simplify_uris_in_tabular(self, data_frame: DataFrame, target_columns: List = None,
+                                 prefix_cc_lookup=True, inplace=True, error_fail=True) -> Dict:
+        """
+            Replace the full URIs by their qname counterparts. Discover the namespaces
+            in the process, if the namespaces are not defined.
+
+        :param error_fail: fail on error or throw exception per data_fame cell
+        :param inplace: indicate whether the current data_frame shall be modified or a new one be created instead
+        :param prefix_cc_lookup:
+        :param target_columns: the target columns to explore;
+                                    Expectation is that these columns exclusively contain only URIs as values
+        :param data_frame: the dataframe to explore
+        :return:  dictionary with newly discovered namespace definitions
+        """
+        if not target_columns:
+            target_columns = []
+
+        for col in target_columns:
+            if col not in data_frame.columns.values.tolist():
+                raise ValueError("The target column not found in the data frame")
+        # get all the string columns
+        obj_columns = data_frame.select_dtypes([np.object]).columns  # [1:]
+        # limit to columns indicated in the self.target_columns
+        obj_columns = filter(lambda x: x in target_columns, obj_columns) if target_columns else obj_columns
+
+        # copy the dataframe if needed
+        result_frame = data_frame if inplace else data_frame.copy(deep=True)
+        for column in obj_columns:
+            #
+            result_frame[column] = result_frame[column].apply(
+                lambda x: self.uri_to_qname(x, prefix_cc_lookup=prefix_cc_lookup, error_fail=error_fail))
+        return result_frame
+
+    def uri_to_qname(self, uri_string, prefix_cc_lookup=True, error_fail=True):
```

Comment:
I  would default `error_fail` to `True`
```python
    def uri_to_qname(self, uri_string, prefix_cc_lookup=True, error_fail=False):
```
it makes more sense from a usecase point.